File: app/src/main/java/com/veganbeauty/app/features/myskin/MySkinFragment.java
package com.veganbeauty.app.features.myskin;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.veganbeauty.app.R;
import com.veganbeauty.app.core.base.RootieFragment;
import com.veganbeauty.app.data.local.LocalJsonReader;
import com.veganbeauty.app.data.local.ProfileSession;
import com.veganbeauty.app.data.local.entities.StoreEntity;
import com.veganbeauty.app.databinding.SkinFragmentHomeBinding;
import com.veganbeauty.app.features.home.BottomNavHelper;

import java.util.List;

public class MySkinFragment extends RootieFragment {

    private SkinFragmentHomeBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = SkinFragmentHomeBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void setupUI(@NonNull View view) {
        String fullName = ProfileSession.getFullName(requireContext());
        if (fullName != null && !fullName.trim().isEmpty()) {
            String[] parts = fullName.trim().split(" ");
            String firstName = parts[parts.length - 1];
            binding.skinGreetingName.setText("Xin chào, " + firstName);
        }

        BottomNavHelper.setup(
                this,
                view,
                R.id.nav_myskin,
                tabId -> {
                    BottomNavHelper.navigate(this, tabId);
                    return null;
                }
        );

        binding.skinBannerContainer.setOnClickListener(v -> openScanFragment());
        binding.skinShortcutScanAi.setOnClickListener(v -> openScanFragment());
        
        binding.skinShortcutProfile.setOnClickListener(v -> {
            if (!ProfileSession.isLoggedIn(requireContext())) {
                BottomNavHelper.showLoginRequiredDialog(requireContext());
                return;
            }
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new com.veganbeauty.app.features.profile.SkinAllergyProfileFragment())
                    .addToBackStack(null)
                    .commit();
        });

        binding.skinShortcutRoutine.setOnClickListener(v -> {
            if (!ProfileSession.isLoggedIn(requireContext())) {
                BottomNavHelper.showLoginRequiredDialog(requireContext());
                return;
            }
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new com.veganbeauty.app.features.routine.SkinTimeRoutineFragment())
                    .addToBackStack(null)
                    .commit();
        });

        binding.skinShortcutReminder.setOnClickListener(v -> {
            if (!ProfileSession.isLoggedIn(requireContext())) {
                BottomNavHelper.showLoginRequiredDialog(requireContext());
                return;
            }
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new com.veganbeauty.app.features.routine.SkinReminderFragment())
                    .addToBackStack(null)
                    .commit();
        });

        binding.skinShortcutWeather.setOnClickListener(v -> {
            if (!ProfileSession.isLoggedIn(requireContext())) {
                BottomNavHelper.showLoginRequiredDialog(requireContext());
                return;
            }
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new com.veganbeauty.app.features.weather.SkinWeatherForecastFragment())
                    .addToBackStack(null)
                    .commit();
        });

        LocalJsonReader jsonReader = new LocalJsonReader(requireContext());
        List<StoreEntity> allStores = jsonReader.getAllStores();
        List<StoreEntity> topStores = allStores.size() > 5 ? allStores.subList(0, 5) : allStores;

        binding.rvSkinStores.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
        SkinStoreAdapter adapter = new SkinStoreAdapter(topStores, store -> {
            BookingFragment bookingFragment = BookingFragment.newInstance(
                    store.getTenCuaHang(),
                    store.getDiaChiDayDu(),
                    ""
            );
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, bookingFragment)
                    .addToBackStack(null)
                    .commit();
        });
        binding.rvSkinStores.setAdapter(adapter);

        binding.skinStoreSeeAll.setOnClickListener(v -> {
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new ChooseBranchFragment())
                    .addToBackStack(null)
                    .commit();
        });

        binding.skinBtnBookingHistory.setOnClickListener(v -> {
            if (!ProfileSession.isLoggedIn(requireContext())) {
                BottomNavHelper.showLoginRequiredDialog(requireContext());
                return;
            }
            getParentFragmentManager().beginTransaction()
                    .setCustomAnimations(
                            android.R.anim.slide_in_left,
                            android.R.anim.fade_out,
                            android.R.anim.fade_in,
                            android.R.anim.slide_out_right
                    )
                    .replace(R.id.main_container, new BookingHistoryFragment())
                    .addToBackStack(null)
                    .commit();
        });
    }

    @Override
    protected void observeViewModel() {
        // Not used yet
    }

    private void openScanFragment() {
        getParentFragmentManager().beginTransaction()
                .setCustomAnimations(
                        android.R.anim.slide_in_left,
                        android.R.anim.fade_out,
                        android.R.anim.fade_in,
                        android.R.anim.slide_out_right
                )
                .replace(R.id.main_container, new SkinScanFragment())
                .addToBackStack(null)
                .commit();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/veganbeauty/app/features/myskin/MySkinFragment.kt
package com.veganbeauty.app.features.myskin

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.veganbeauty.app.R
import com.veganbeauty.app.core.base.RootieFragment
import com.veganbeauty.app.data.local.ProfileSession
import com.veganbeauty.app.databinding.SkinFragmentHomeBinding
import com.veganbeauty.app.utils.NavAppUtils
import androidx.recyclerview.widget.LinearLayoutManager
import com.veganbeauty.app.data.local.LocalJsonReader

class MySkinFragment : RootieFragment() {

    private var _binding: SkinFragmentHomeBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = SkinFragmentHomeBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun setupUI(view: View) {
        // Cài đặt tên chào người dùng
        val fullName = ProfileSession.getFullName(requireContext())
        if (!fullName.isNullOrBlank()) {
            val firstName = fullName.trim().split(" ").lastOrNull() ?: fullName
            binding.skinGreetingName.text = "Xin chào, $firstName"
        }

        // Cài đặt bottom nav
        NavAppUtils.setupNavApp(this, view, R.id.nav_myskin)

        // Banner và shortcut Quét da AI → mở SkinScanFragment
        binding.skinBannerContainer.setOnClickListener { openScanFragment() }
        binding.skinShortcutScanAi.setOnClickListener { openScanFragment() }
        binding.skinShortcutProfile.setOnClickListener {
            android.widget.Toast.makeText(context, "Hồ sơ da — đang phát triển", android.widget.Toast.LENGTH_SHORT).show()
        }
        binding.skinShortcutRoutine.setOnClickListener {
            android.widget.Toast.makeText(context, "Routine chăm da — đang phát triển", android.widget.Toast.LENGTH_SHORT).show()
        }
        binding.skinShortcutReminder.setOnClickListener {
            android.widget.Toast.makeText(context, "Nhắc lịch — đang phát triển", android.widget.Toast.LENGTH_SHORT).show()
        }
        binding.skinShortcutWeather.setOnClickListener {
            android.widget.Toast.makeText(context, "Skin Weather — đang phát triển", android.widget.Toast.LENGTH_SHORT).show()
        }

        // Setup RecyclerView cho cửa hàng
        val jsonReader = LocalJsonReader(requireContext())
        val allStores = jsonReader.getAllStores()
        val topStores = allStores.take(5)
        
        binding.rvSkinStores.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        val adapter = SkinStoreAdapter(topStores) { store ->
            val bookingFragment = BookingFragment.newInstance(
                store.storeName,
                store.address,
                store.imageUrl
            )
            parentFragmentManager.beginTransaction()
                .setCustomAnimations(
                    android.R.anim.slide_in_left,
                    android.R.anim.fade_out,
                    android.R.anim.fade_in,
                    android.R.anim.slide_out_right
                )
                .replace(R.id.main_container, bookingFragment)
                .addToBackStack(null)
                .commit()
        }
        binding.rvSkinStores.adapter = adapter

        binding.skinStoreSeeAll.setOnClickListener {
            parentFragmentManager.beginTransaction()
                .setCustomAnimations(
                    android.R.anim.slide_in_left,
                    android.R.anim.fade_out,
                    android.R.anim.fade_in,
                    android.R.anim.slide_out_right
                )
                .replace(R.id.main_container, ChooseBranchFragment())
                .addToBackStack(null)
                .commit()
        }

        // Lịch sử lịch đặt
        binding.skinBtnBookingHistory.setOnClickListener {
            parentFragmentManager.beginTransaction()
                .setCustomAnimations(
                    android.R.anim.slide_in_left,
                    android.R.anim.fade_out,
                    android.R.anim.fade_in,
                    android.R.anim.slide_out_right
                )
                .replace(R.id.main_container, BookingHistoryFragment())
                .addToBackStack(null)
                .commit()
        }
    }

    override fun observeViewModel() {
        // Not used yet
    }

    private fun openScanFragment() {
        parentFragmentManager.beginTransaction()
            .setCustomAnimations(
                android.R.anim.slide_in_left,
                android.R.anim.fade_out,
                android.R.anim.fade_in,
                android.R.anim.slide_out_right
            )
            .replace(R.id.main_container, SkinScanFragment())
            .addToBackStack(null)
            .commit()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
